### ISSUE TITLE
Add ability to set the timezone along with dateFormat

### ISSUE THREAD

**Issue by leothekim** [ktoso/maven-git-commit-id-plugin#204]
We report the build time that's generated in `git.properties` in our webapp, and we'd like to be able to set the `ZoneId` to UTC for the formatted datetime to make it consistent with other timestamps we report. Seems like this would just be adding a `setTimeZone()` call based on a `<timeZone>` field.

**Comment by TheSnoozer** [comment]
Hi @leothekim,
thanks for reporting your issue here.
It sounds reasonable to add this to the plugin.

I would add something like
``smf.setTimeZone(java.util.TimeZone.getTimeZone("UTC"));`` whereas UTC is the parameter you can configure in the plugin's configuration.

By the way you can work around this by adding ``-Duser.timezone=PST`` to your build options or adding the user.timezone to your pom's properties.

Also I would say as long as those properties are present the plugin's configuration should be ignored (so ``-Duser.timezone=PST`` would override the plugin's timeZone config)

**Comment by leothekim** [comment]
Thanks @TheSnoozer for looking into this! I tried `-Duser.timezone=UTC` and its equivalents in various places of pom.xml (in the configuration stanza for this plugin, in the compiler plugin, etc), but couldn't get it to change the timezone of the build time. Didn't look at it too long though, so I probably just missed something.

**Comment by TheSnoozer** [comment]
According to https://docs.oracle.com/javase/7/docs/api/java/util/TimeZone.html:
```
For compatibility with JDK 1.1.x, some other three-letter time zone IDs (such as "PST", "CTT", "AST") are also supported. However, their use is deprecated because the same abbreviation is often used for multiple time zones (for example, "CST" could be U.S. "Central Standard Time" and "China Standard Time"), and the Java platform can then only recognize one of them.
```
this will go in the docs....

**Comment by TheSnoozer** [comment]
Hi @leothekim,
thanks for the suggestion.
Feature will be shipped in 2.2.0.

**Comment by leothekim** [comment]
Thank you, @TheSnoozer! Good call on the TimeZone issue too!
